BindingContext/BPoP.md

```diff
@@ -0,0 +1,313 @@
+    ...
+    let bPoP: boolean = false;
+    if(cookieBindingPolicy){
+        bPoP = true;
+    }
+    const loginUrl = generateLoginUrl(.., bPop);
+    navigateToLoginUrl(Url);
+}
+
+// internal API to process the server response
+processTokenResponse(serverResponseParams): AuthenticationResult {
+
+    // process regular token response
+    accessToken = ..
+
+    ...
+    // process bPop params
+    const bPoPParams : {
+        bPop: serverResponseParams.bPop,
+        nonce?: serverResponseParams.bPopNonce ? serverResponseParams.bPop-nonce : undefined,
+        subdomains: serverResponseParams.subdomains? serverResponseParams.subdomains : false,
+        algs: serverResponseParams.algs,
+        refreshIn: serverResponseParams.refreshIn? serverResponseParams.refreshIn : 300,
+        expiresIn: serverResponseParams.expiresIn? serverResponseParams.expiresIn: 300
+    };
+
+    return generateAuthenticationResult(tokenParams, bPopParams);
+}
+
+/**
+ * API to fetch tokens to a SPA
+ */
+acquireToken(request) {
+    ...
+    // tokenResponse
+    const serverResponse: ServerResponseParams = getToken(.., cookieBindingPolicy: boolean = true);
+    return processTokenResponse(serverResponse);
+}
+
+```
+
+### Browser BPoP proofs
+
+A BPoP proof is a signed [CWT](https://www.rfc-editor.org/rfc/rfc8392.html). A BPoP proof demonstrates to the server that the client holds the private key that was used to sign the BPoP proof CWT. This enables websites to bind issued browser storage artifacts (e.g cookies) to the corresponding public key and to verify the key binding of all artifacts they receive, which prevents said artifacts from being used by any entity that does not have access to the private key.
+
+The COSE header of a BPoP CWT MUST contain at least the following parameters:
+
+- `typ` with value `bpop+jwt`
+- `alg` a digital signature algorithm identifier chosen from the list indicated by the server.
+- `jwk` representing the public key chosen by the client, in JSON Web Key (JWK) format [RFC7517](https://datatracker.ietf.org/doc/html/rfc7517).
+
+The payload of a BPoP CWT MUST contain at least the following claims:
+
+- `iat` creation timestamp of the CWT
+- `hth` with value equal to the host value of the http request.
+
+Following a `BPoP-Nonce` header, the BPoP proof must also contain a claim `nonce` with value equal to that header.
+
+The client sends a BPoP proof on future HTTP requests.
+
+```
+BPoP: eyJ0eXAiOiJicG9wK2p3dCIsImFsZyI6IkVTMjU2IiwiandrIjp7Imt0eSI6IkV
+   DIiwieCI6Imw4dEZyaHgtMzR0VjNoUklDUkRZOXpDa0RscEJoRjQyVVFVZldWQVdCR
+   nMiLCJ5IjoiOVZFNGpmX09rX282NHpiVFRsY3VOSmFqSG10NnY5VERWclUwQ2R2R1J
+   EQSIsImNydiI6IlAtMjU2In19.eyJodGgiOiJzZXJ2ZXIuZXhhbXBsZS5jb20iLCJp
+   YXQiOjE1NjIyNjI2MTZ9.2-GxA6T8lP4vfrg8v-FdWP0A0zdrj8igiMLvqRMUvwnQg
+   4PtFLbdLXiOSsX0x7NVY-FNyJK70nfbV37xRZT3Lg
+```
+
+Note: This example can be a CWT instead of a JWT.
+
+The client is expected to cache BPoP proofs and re-use them, until rejected by the server.
```

Comment:
Should this include "until expires-in" has been reached in addition to "until rejected by the server"

Reply:
Added.